modules/attributes/pages/document-attributes-reference.adoc

```diff
@@ -36,6 +37,229 @@ Otherwise, the assignment won't have any effect on the document.
 If an attribute is not marked as _Header Only_, it can be set anywhere in the document, assuming the attribute is not locked by the API or CLI.
 However, changing an attribute only affects behavior for content that follows the assignment (in document order).
 
+== Intrinsic attributes
+
+Intrinsic attributes are set automatically by the processor.
+These attributes provide information about the document being processed (e.g., `docfile`), the security mode under which the processor is running (e.g., `safe-mode-name`), and information about the user's environment (e.g., `user-home`).
+
+Many of these attributes are read only, which means they can't be modified (with some exceptions).
+Attributes which are not are marked as modifiable.
+Attributes marked as both modifiable and API/CLI Only can only be set from the API or CLI.
+All other attributes marked as modifiable must be set by the end of the header (i.e., Header Only).
```

Comment:
Maybe we should tell "in the document header" instead of "by the end of the header". The term "Document Header" seems well defined so it might be easier to understand if we use "in the document header".

Reply:
I've been introducing the phrase "by the end of the header" since that more accurately describes what's really going on. We can't just say "Document Header" because that implies you must set it in the Document Header itself, which is not accurate.